webapp/store/page/home/homeSlice.ts

```diff
@@ -0,0 +1,30 @@
+/* eslint-disable no-param-reassign */
+
+import { createSlice, Reducer } from '@reduxjs/toolkit'
+
+import { HomeCountriesFilterActions } from '@webapp/store/page/home/actionTypes'
+import { AppActions } from '@webapp/store/app'
+import { HomeState } from './HomeStateType'
+
+const initialState: HomeState = {}
+
+export const homeSlice = createSlice({
+  name: 'home',
+  initialState,
+  reducers: {
+    updateCountriesFilter: (_state: HomeState, action: HomeCountriesFilterActions): HomeState => {
+      return {
+        countriesFilter: action.payload,
+      }
+    },
+  },
+  extraReducers: (builder) => {
+    builder.addCase(AppActions.updateCountryIso, () => initialState)
+  },
+})
+
+export const HomeActions = {
+  ...homeSlice.actions,
+}
+
+export default homeSlice.reducer as Reducer<HomeState>
```

Comment:
is this cast needed? isn't createSlice typed already ?

Reply:
Taken from RTK example, removed